staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go

```diff
@@ -1209,6 +1221,22 @@ func seedMultiLevelData(ctx context.Context, store storage.Interface) (string, [
 		if err != nil {
 			return "", nil, fmt.Errorf("failed to create object: %w", err)
 		}
+		// For barFirst, we first create it with key /pods/first/bar
+		// and then we update it by changing its spec.nodeName. The
+		// point of doing this is to be able to test that if a pod
+		// with key /pods/first/bar is in fact returned, the returned
+		// pod is the updated one (i.e. with spec.nodeName changed).
+		if i == 0 {
+			preset[i].storedObj = &example.Pod{}
+			if err := store.GuaranteedUpdate(ctx, ps.key, preset[i].storedObj, true, nil,
+				func(input runtime.Object, _ storage.ResponseMeta) (output runtime.Object, ttl *uint64, err error) {
+					pod := input.(*example.Pod)
+					pod.Spec.NodeName = "fakeNode"
+					return pod.DeepCopyObject(), nil, nil
```

Comment:
First do a DeepCopy, and only then modify the Spec.NodeName.

Reply:
ah yes, missed that
done, thanks!